pkgs/development/python-modules/nicegui/default.nix

```
@@ -0,0 +1,123 @@
+{
+  lib,
+  aiofiles,
+  aiohttp,
+  buildPythonPackage,
+  certifi,
+  pkgs,
+  docutils,
+  fastapi,
+  fetchFromGitHub,
+  httpx,
+  ifaddr,
+  itsdangerous,
+  jinja2,
+  libsass,
+  markdown2,
+  matplotlib,
+  orjson,
+  pandas,
+  plotly,
+  poetry-core,
+  pyecharts,
+  pygments,
+  pytest-asyncio,
+  pytest-selenium,
+  pytestCheckHook,
+  python-multipart,
+  python-socketio,
+  pythonOlder,
+  pywebview,
+  requests,
+  setuptools,
+  typing-extensions,
+  urllib3,
+  uvicorn,
+  vbuild,
+  watchfiles,
+  webdriver-manager,
+}:
+
+buildPythonPackage rec {
+  pname = "nicegui";
+  version = "2.5.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "zauberzeug";
+    repo = "nicegui";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-oT191QVpvE5xszgBFt3o4A2hU50zmzPUywmAQuKZ5OE=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "setuptools>=30.3.0,<50" "setuptools"
```

```suggestion
      --replace-fail "setuptools>=30.3.0,<50" ""
```
same here
https://github.com/search?q=repo%3Azauberzeug%2Fnicegui%20setuptools&type=code